src/utils/hooks/useGetResponsesByCode.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react'
+import {
+  Application,
+  ApplicationResponse,
+  ApplicationSection,
+  useGetApplicationQuery,
+  GetApplicationQuery,
+} from '../generated/graphql'
+import { ResponsesByCode } from '../types'
+
+interface useLoadApplicationProps {
+  serialNumber: string
+}
+
+const useGetResponsesByCode = (props: useLoadApplicationProps) => {
+  const { serialNumber } = props
+  const [responsesByCode, setResponsesByCode] = useState({})
+  const [error, setError] = useState('')
+  const { data, loading: apolloLoading, error: apolloError } = useGetApplicationQuery({
+    variables: { serial: serialNumber },
+  })
+
+  useEffect(() => {
+    const error = checkForApplicationErrors(data)
+
+    if (error) {
+      setError(error)
+      return
+    }
+
+    const applicationResponses = data?.applications?.nodes[0]?.applicationResponses
+      .nodes as ApplicationResponse[]
+
+    const currentResponses = {} as ResponsesByCode
```

Comment:
Oh, that's interesting. I usually initialise the variable like this: `const currentResponses: ResponsesByCode = {}`, but I liked the way this is done it's more declarative : )

Reply:
Ha ha, so do I actually -- I guess I'd just got used to seeing your `as` style a lot recently. 😄